### PR TITLE
Add Flow Types

### PR DESCRIPTION
**Pull request by jgcmarins**
I've configured Flow and fixed some errors. But there are 2 errors that I am not finding solution:

```
src/generateSVGFont.js:44
 44:     glyph.metadata = {
               ^^^^^^^^ property `metadata`. Property not found in
 44:     glyph.metadata = {
         ^^^^^ ReadStream

src/runCLI.js:45
                                           v
 45: export default async function execute({
 46:   fontName = 'MyFont',
 47:   dest = 'dist',
 48:   iconsGlob = './icons/**/*.svg',
 49: } = {}) {
     ^ destructuring. Missing annotation


Found 2 errors
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,11 +18,12 @@
...
   },
   "devDependencies": {
-    "babel-cli": "^6.9.0",
+    "babel-cli": "^6.24.1",
     "babel-preset-es2015": "^6.9.0",
     "babel-preset-flow": "^6.23.0",
     "babel-preset-stage-0": "^6.24.1",
-    "babel-register": "^6.9.0"
+    "babel-register": "^6.9.0",
+    "flow-bin": "^0.47.0"
```
Comment (sibelius) [inline package.json:27]: add `"flow": "flow"` on scripts part

**Comment by sibelius** [comment]
try this:

```jsx
type CliOptions = {
  fontName: string,
  dest: string, 
  iconsGlob: string, 
}
export default async function execute({
  fontName = 'MyFont',
  dest = 'dist',
  iconsGlob = './icons/**/*.svg',
} = {}: CliOptions) {
```

**Comment by sibelius** [comment]
could you also add flow-typed https://github.com/flowtype/flow-typed?

File: src/generateSVGFont.js [hunk 2]
```diff
@@ -55,7 +55,7 @@ const generateSvgFontStream = (options, done) => {
  * Promisify generateSvgFontStream
  * @param options
  */
-const generateSvgFontAsync = options =>
+const generateSvgFontAsync = (options: Object) =>
```
Comment (sibelius) [inline src/generateSVGFont.js:58]: can you be more specific?

```jsx
type SvgFontOptions = {
   fontName: string,
    fontHeight: number,
    descent: ...,
    normalize: ...,
    round: ...,
}
```

File: src/svgToTTF.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ import svg2ttf from 'svg2ttf';
  * @param options
  * @returns {Buffer}
  */
-const generateTTFfromSVGFont = (svgFont, options) => {
+const generateTTFfromSVGFont = (svgFont: string, options: Object) => {
```
Comment (sibelius) [inline src/svgToTTF.js:12]: be more specific on `options`

**Comment by jgcmarins** [comment]
- [x] add `"flow": "flow"` to scripts in `package.json`
- [x]  add `flow-typed`
- [x] `type CliOptions`
- [ ] be more specific on `options` type

@sibelius , I am having some troubles defining a type for `options`.
I've followed [this](https://github.com/nfroidure/svgicons2svgfont#api), but I guess that the problem is me. I don't know enough JS to solve. Do you have any similar example?

**Review by sibelius — APPROVED** [review]
Great job, we can improve Flow support in another PR

**Comment by jgcmarins** [comment]
Thanks :)
You helped me a lot!

**Pull request merged**